packages/multichain-transactions-controller/src/MultichainTransactionsController.ts

```diff
@@ -404,45 +400,64 @@ export class MultichainTransactionsController extends BaseController<
+          const chain = tx.chain as string;
 
-        // Uses a `Map` to deduplicate transactions by ID, ensuring we keep the latest version
-        // of each transaction while preserving older transactions and transactions from other accounts.
-        // Transactions are sorted by timestamp (newest first).
-        const transactions = new Map();
+          if (!updatedTransactions[accountId][chain]) {
+            updatedTransactions[accountId][chain] = [];
+          }
 
-        oldTransactions.forEach((tx) => {
-          transactions.set(tx.id, tx);
-        });
-
-        filteredNewTransactions.forEach((tx) => {
-          transactions.set(tx.id, tx);
+          updatedTransactions[accountId][chain].push(tx);
           transactionsToPublish.push(tx);
         });
 
-        // Sorted by timestamp (newest first). If the timestamp is not provided, those
-        // transactions will be put in the end of this list.
-        updatedTransactions[accountId] = Array.from(transactions.values()).sort(
-          (a, b) => (b.timestamp ?? 0) - (a.timestamp ?? 0),
+        Object.entries(updatedTransactions[accountId]).forEach(
+          ([chain, chainTransactions]) => {
+            // Account might not have any transactions yet, so use `[]` in that case.
+            const oldTransactions =
+              this.state.nonEvmTransactions[accountId]?.[chain]?.transactions ??
+              [];
+
+            // Uses a `Map` to deduplicate transactions by ID, ensuring we keep the latest version
+            // of each transaction while preserving older transactions and transactions from other accounts.
+            // Transactions are sorted by timestamp (newest first).
+            const transactions = new Map();
+
+            oldTransactions.forEach((tx) => {
+              transactions.set(tx.id, tx);
+            });
+
+            chainTransactions.forEach((tx) => {
+              transactions.set(tx.id, tx);
+            });
+
+            // Sorted by timestamp (newest first). If the timestamp is not provided, those
+            // transactions will be put in the end of this list.
+            updatedTransactions[accountId][chain] = Array.from(
+              transactions.values(),
+            ).sort((a, b) => (b.timestamp ?? 0) - (a.timestamp ?? 0));
+          },
         );
       },
     );
 
     this.update((state) => {
-      Object.entries(updatedTransactions).forEach(
-        ([accountId, transactions]) => {
-          state.nonEvmTransactions[accountId] = {
-            ...state.nonEvmTransactions[accountId],
+      Object.entries(updatedTransactions).forEach(([accountId, chainsData]) => {
+        if (!state.nonEvmTransactions[accountId]) {
+          state.nonEvmTransactions[accountId] = {};
+        }
+
+        Object.entries(chainsData).forEach(([chain, transactions]) => {
+          state.nonEvmTransactions[accountId][chain] = {
+            ...state.nonEvmTransactions[accountId][chain],
```

Comment:
Do we really need that one now? 🤔 We should already have the `oldTransactions` as part of `transactions` no?

Also, anything from `state.nonEvmTransactions[accountId][chain]` would just get overriden here.